### ISSUE TITLE
"Error resolving project artifact" unless comes from maven central

### ISSUE THREAD

**Issue by OliverMatz** [spdx/spdx-maven-plugin#84]
I have a tiny maven project that uses the maven repository `https://jcenter.bintray.com/` which is needed to resolve the dependency `org.jfrog.security:jfrog-crypto`.

It compiles and runs without problems, which (I think) proves that the setup is correct.

However, the spdx-maven-plugin fails with the error message mentioned below.

I had expected the spdx-maven-plugin to download `jfrog-crypto-4.9.0.pom` from `https://jcenter.bintray.com/` rather than maven central.

The maven project can be found in the attachment.

```
[INFO] org.spdx.maven.CreateSpdxMojo - Creating SPDX File C:\dvdv\spdx-maven-plugin-minimal-0\target\site\org.example_spdx-maven-plugin-minimal-1.0-SNAPSHOT.spdx.json
Downloading from central: https://repo.maven.apache.org/maven2/org/jfrog/security/jfrog-crypto/4.9.0/jfrog-crypto-4.9.0.pom
[ERROR] org.spdx.maven.CreateSpdxMojo - Maven Project Build Error creating SPDX package for dependency artifact ID jfrog-crypto:Error resolving project artifact: Could not find artifact org.jfrog.security:jfrog-crypto:pom:4.9.0 in central (https://repo.maven.apache.org/maven2) for project org.jfrog.security:jfrog-crypto:jar:4.9.0
[WARNING] org.spdx.maven.CreateSpdxMojo - Error creating SPDX package for dependency artifact ID jfrog-crypto.  A minimal SPDX package will be created.
Downloading from central: https://repo.maven.apache.org/maven2/org/jfrog/common/jfrog-utils/4.9.0/jfrog-utils-4.9.0.pom
[ERROR] org.spdx.maven.CreateSpdxMojo - Maven Project Build Error creating SPDX package for dependency artifact ID jfrog-utils:Error resolving project artifact: Could not find artifact org.jfrog.common:jfrog-utils:pom:4.9.0 in central (https://repo.maven.apache.org/maven2) for project org.jfrog.common:jfrog-utils:jar:4.9.0
[WARNING] org.spdx.maven.CreateSpdxMojo - Error creating SPDX package for dependency artifact ID jfrog-utils.  A minimal SPDX package will be created.
```
[spdx-maven-plugin-minimal-0.zip](https://github.com/spdx/spdx-maven-plugin/files/10990115/spdx-maven-plugin-minimal-0.zip)

**Comment by goneall** [comment]
@OliverMatz Thanks for reporting the issue.  I'm not quite sure what the mechanism is for downloading the artifacts - I thought it was using the same mechanism used by Maven when doing other build.  I'll look into it over the weekend and update this issue with any finding.

**Comment by goneall** [comment]
Just an update - I've isolated the problem to the following call to `MavenProjectBuilder.build(...)` which does not seem to use the correct repository when finding the dependencies:

https://github.com/spdx/spdx-maven-plugin/blob/d711f01cdb09e7cbab08c71b117d0569b119d402/src/main/java/org/spdx/maven/SpdxDependencyInformation.java#L309

It looks like the `MavenSession.getProjectBuildingRequest()` does not have the jfrog repository in it's list of external repositories.

I'll see if I can find out why...

**Comment by goneall** [comment]
Thanks @OliverMatz for catching the error (I should have checked the output more carefully).

I created PR #86 which seems to fix the issue but generates yet another error which I need to investigate.  Let me know any thoughts.

**Comment by OliverMatz** [comment]
The plugin version built from https://github.com/spdx/spdx-maven-plugin/tree/wrongdep now works for my project.
I must say, though, that I have a couple of questions, but that is probably because I am new to spdx.

**Comment by goneall** [comment]
@OliverMatz I'll go ahead and merge the PR in.  I created a separate issue to track resolving the circular dependencies mentioned in the PR comments.

For any SPDX questions, feel free to post questions to the[ SPDX tech email list](https://lists.spdx.org/g/spdx-tech), post an issue to the [SPDX Spec repo](https://github.com/spdx/spdx-spec/issues), or email me at gary at sourceuditor.com.

**Comment by goneall** [comment]
Fixed with PR #86
